public/app/plugins/datasource/mixed/MixedDataSource.ts

```diff
@@ -105,7 +105,19 @@ export class MixedDatasource extends DataSourceApi<DataQuery> {
       observables.push(observable);
     }
 
-    return merge(...observables);
+    return forkJoin(observables).pipe(
```

Comment:
Does forkJoin guarantee order?

Reply:
Yes, it will return the results in the order of the passed observables.

https://indepth.dev/learn-to-combine-rxjs-sequences-with-super-intuitive-interactive-diagrams/#forkjoin

Reply:
woohoo 🍾